client/src/elements/schema-editor/schema-editor-dialog.js

```diff
@@ -0,0 +1,59 @@
+import { bindable, inject } from "aurelia-framework";
+import { DialogService } from "aurelia-dialog";
+import { I18N } from "aurelia-i18n";
+import { getType, isRequired } from "./helpers.js";
+import { SchemaEditorDialogImplementation } from "../../dialogs/schema-editor-dialog.js";
+
+@inject(DialogService, I18N)
+export class SchemaEditorDialog {
+  @bindable
+  schema;
+  @bindable
+  data;
+  @bindable
+  change;
+  @bindable
+  notifications;
+  @bindable
+  showNotifications;
+
+  constructor(dialogService, i18n) {
+    this.getType = getType;
+    this.isRequired = isRequired;
+    this.dialogService = dialogService;
+    this.i18n = i18n;
+  }
+
+  options = {};
+
+  schemaChanged() {
+    this.applyOptions();
+  }
+
+  applyOptions() {
+    if (this.schema.hasOwnProperty("Q:options")) {
+      this.options = Object.assign(this.options, this.schema["Q:options"]);
+    }
+  }
+
+  async open() {
+    const dialog = await this.dialogService.open({
+      viewModel: SchemaEditorDialogImplementation,
+      model: {
+        schema: this.schema,
+        data: this.data,
+        change: this.change,
```

Comment:
As far as I understand it works like this now:
data and change get passed through to the schema-editor in the dialog. As these are references, the data and form outside of the dialog change and data gets persisted if autosave is running.

I would do a clone of the data (`JSON.parse(JSON.stringify(this.data))`) here to take the data in that is already defined but not change the referenced data object but change the reference when the save button was clicked.

The `change` could just be undefined here, and then called after the dialog closed and the data is stored.